src/mgmt/srv_pool.c

```diff
@@ -573,47 +574,66 @@ int
 ds_mgmt_destroy_pool(uuid_t pool_uuid, const char *group, uint32_t force)
 {
 	struct mgmt_svc	*svc;
-	int		rc;
+	d_rank_list_t	*psvcranks;
+	int		 rc;
 
-	/* TODO check metadata about the pool's existence?
-	 *      and check active pool connection for "force"
-	 */
 	D_DEBUG(DB_MGMT, "Destroying pool "DF_UUID"\n", DP_UUID(pool_uuid));
 
 	rc = ds_mgmt_svc_lookup_leader(&svc, NULL /* hint */);
 	if (rc != 0)
 		goto out;
 
+	/* Get pool service ranks before pool_destroy_prepare() that changes
+	 * pr_state to destroying.
```

Comment:
Could you elaborate this a bit? Sorry, I'm being slow...

If the pr_state check in pool_get_svc_ranks is the problem, could we change it, for instance, to

    } else if (rec->pr_state != POOL_READY &&
               rec->pr_state != POOL_DESTROYING)

Then we could get the ranks and do the evict after pool_destroy_prepare.

Reply:
For now in most recent patch I've adopted the suggested change to pool_get_svc_ranks(). Trying to remember exact scenarios causing me to perform those steps before pool_destroy_prepare(). One of them may have been:
call 1 (force=false): pool_destroy_prepare() leaves pr_state = POOL_DESTROYING, meanwhile the eviction check fails due to open connections/handles on the pool - so the pool is not actually destroyed. No problem (yet).
call 2 (force=true): the call to pool_get_svc_ranks() fails because of the state, where a caller should expect the overall force destroy procedure to succeed.

Another consideration: should the pool also disallow any further connections when the  attempt to destroy it (with force=false) fails due to open handles? The pool service code added does not change connectable in this case (i.e., the pool can continue to accept new connections). But with the pool's pr_state=POOL_DESTROYING (I need to test this) - will the pool effectively be unable to accept any more connections?

Reply:
Indeed. Sorry, I missed these valid concerns.